ultraplot/tests/test_tickers.py

```diff
@@ -0,0 +1,576 @@
+    assert zerotrim is True
+
+    # Case 2: Explicit precision and zerotrim
+    precision, zerotrim = _default_precision_zerotrim(precision=3, zerotrim=False)
+    assert precision == 3
+    assert zerotrim is False
+
+
+def test_index_locator_tick_values():
+    from ultraplot.ticker import IndexLocator
+
+    locator = IndexLocator(base=2, offset=1)
+    ticks = locator.tick_values(0, 10)
+    assert np.array_equal(ticks, [1, 3, 5, 7, 9])
+
+
+def test_discrete_locator_call():
+    from ultraplot.ticker import DiscreteLocator
+
+    locator = DiscreteLocator(locs=[0, 1, 2, 3, 4])
+    ticks = locator()
+    assert np.array_equal(ticks, [0, 1, 2, 3, 4])
+
+
+def test_discrete_locator_set_params():
+    from ultraplot.ticker import DiscreteLocator
+
+    locator = DiscreteLocator(locs=[0, 1, 2, 3, 4])
+    locator.set_params(steps=[1, 2], nbins=3, minor=True, min_n_ticks=2)
+
+    assert np.array_equal(locator._steps, [1, 2, 10])
+    assert locator._nbins == 3
+    assert locator._minor is True
+    assert locator._min_n_ticks == 2
+
+
+def test_discrete_locator_tick_values():
+    from ultraplot.ticker import DiscreteLocator
+
+    # Create a locator with specific locations
+    locator = DiscreteLocator(locs=[0, 1, 2, 3, 4, 5, 6, 7, 8, 9])
+    locator.set_params(steps=[1, 2], nbins=5, minor=False, min_n_ticks=2)
+
+    ticks = locator.tick_values(None, None)
+
+    assert np.array_equal(ticks, list(range(10)))
+
+
+@pytest.mark.parametrize(
+    "value, string, expected",
+    [
+        (1e-10, "0.0", True),  # Case 1: Small number close to zero
+        (1000, "1000", False),  # Case 2: Large number
+    ],
+)
+def test_auto_formatter_fix_small_number(value, string, expected):
+    from ultraplot.ticker import AutoFormatter
+
+    formatter = AutoFormatter()
+    result = formatter._fix_small_number(value, string)
+    if expected:
+        assert result != string
+    else:
+        assert result == string
+
+
+@pytest.mark.parametrize(
+    "start_date, end_date",
+    [
+        (
+            cftime.date2num(
+                cftime.DatetimeGregorian(2000, 1, 1), "days since 2000-01-01"
+            ),
+            cftime.date2num(
+                cftime.DatetimeGregorian(2020, 1, 1), "days since 2000-01-01"
+            ),
+        ),
+        # Case 2: Monthly resolution
+        (
+            cftime.date2num(
+                cftime.DatetimeGregorian(2000, 1, 1), "days since 2000-01-01"
+            ),
+            cftime.date2num(
+                cftime.DatetimeGregorian(2000, 12, 31), "days since 2000-01-01"
+            ),
+        ),
+    ],
+)
+def test_auto_datetime_locator_tick_values(start_date, end_date):
+    from ultraplot.ticker import AutoCFDatetimeLocator
+
+    locator = AutoCFDatetimeLocator(calendar="gregorian")
+    import cftime
+
+    ticks = locator.tick_values(start_date, end_date)
+    assert len(ticks) > 0  # Ensure ticks are generated
+
+
+@pytest.mark.parametrize(
```

Comment:
There are no test cases here that are expected to raise. Did you intend to add them? Should we add some?

Reply:
Removed the block.